python/ray/train/torch/train_loop_utils.py

```diff
@@ -63,11 +69,64 @@ def get_device() -> Union[torch.device, List[torch.device]]:
         >>> # ray.get_gpu_ids() == [4,5]
         >>> # torch.cuda.is_available() == True
         >>> # get_device() == torch.device("cuda:4")
+
+
+        >>> # You can move model to device by:
+        >>> # model.to(ray.train.torch.get_device())
+        >>> #
+        >>> # instead of manually checking the device type:
+        >>> # model.to("cuda" if torch.cuda.is_available() else "cpu")
     """
     from ray.air._internal import torch_utils
 
     record_extra_usage_tag(TagKey.TRAIN_TORCH_GET_DEVICE, "1")
-    return torch_utils.get_device()
+    return torch_utils.get_devices()[0]
+
+
+@PublicAPI(stability="alpha")
+def get_devices() -> List[torch.device]:
+    """Gets the correct torch device list configured for this process.
+
+    Assumes that `CUDA_VISIBLE_DEVICES` is set and is a
+    superset of the `ray.get_gpu_ids()`.
+
+
+    Examples:
+
+        .. testcode::
+            :skipif: True
+
+            import torch
+            import os
+            import ray
+            import ray.train
+
+            # Example: Launched 2 workers, each with 1 GPU
+            def train_func():
+                os.environ["CUDA_VISIBLE_DEVICES"] == "3,4"
+                ray.get_gpu_ids() == [3]
+                torch.cuda.is_available() == True
+                get_devices() == [torch.device("cuda:0")]
+
+            # Example: Launched 5 workers, each with 1 GPU
+            def train_func():
+                os.environ["CUDA_VISIBLE_DEVICES"] == "0,1,2,3,4"
+                ray.get_gpu_ids() == [4]
+                torch.cuda.is_available() == True
+                get_devices() == [torch.device("cuda:4")]
+
+            # Example: Launched 3 workers, each with 2 GPUs
+            def train_func():
+                os.environ["CUDA_VISIBLE_DEVICES"] == "0,1,2,3,4,5"
+                ray.get_gpu_ids() == [4,5]
+                torch.cuda.is_available() == True
+                get_devices() == [torch.device("cuda:4"), torch.device("cuda:5")]
+    """
+
+    from ray.air._internal import torch_utils
+
+    record_extra_usage_tag(TagKey.TRAIN_TORCH_GET_DEVICE, "1")
```

Comment:
This should have a new `TRAIN_TORCH_GET_DEVICES` key.

Reply:
Ah previously Justin mentioned we can use a single Key for these two APIs. But now I think it makes more sense to have a separate one to get more accurate telemetry data.